uwblueprint/community-fridge-kw
# feat: make my account editable

## Brief description. What is this change? 

### [Update My Account Page to be Editable](https://www.notion.so/uwblueprintexecs/Update-My-Account-Page-to-be-Editable-44852ec1a17743af8db408cc913b442b)



## Implementation description. How did you make this change?
* Edited Inputs in My Account to be editable 
* Added submittable formValues for My Account fields
* Added formErrors for validation
* Added a confirmCancelEditModal component
* Save values to database + update local storage with new values
* Also ran `yarn fix` for linting a bunch of files




https://user-images.githubusercontent.com/37950626/150910686-949a4333-8ef5-4e94-849f-56168d1d5f35.mov

## Steps to test
1. Login and go to My Account page
2. Click 'Edit' -> all fields except email should be visibly editable with a text box
3. Click 'X' to cancel changes -> confirmation modal should appear (clicking X goes back to edit mode and 'Discard changes' should reset the form to the original changes
4. Clear the fields.
5. Click 'Save changes' 
6. There should be errors saying the corresponding empty fields are required
7. Fill in the fields and click 'Save changes'. 
8. Refresh the page
9. The values should be updated correctly. You can also check pgadmin to see if your field is correctly updated in the `donors` and `users` tables.
10. Clicking 'Back' goes to Dashboard


## Checklist
